src/runtime/virtcontainers/qemu_amd64.go

```diff
@@ -140,6 +140,18 @@ func newQemuArch(config HypervisorConfig) (qemuArch, error) {
 		}
 	}
 
+	if config.SGXEPCSize != 0 {
+		q.sgxEPCSize = config.SGXEPCSize
+		if q.qemuMachine.Options != "" {
+			q.qemuMachine.Options += ","
+		}
+		// the same way we did with cloud-hypervisor,
+		// qemu sandboxes will only support one EPC per sandbox
+		// this is because there is only one annotation (sgx.intel.com/epc)
+		// to specify the size of the EPC.
```

Comment:
Can you help clarify -- we could have multiple EPC in a VM, but we aren't supporting it because of how we defined the sgx annotations?

I assume we receive this annotation in a container spec? or, can you please clarify?  Who defines this/these annotations to set this limitation?

Thanks -- just want to make sure I understand!

Reply:
The annotation comes from the SGX device plugin (or it can be specified as the pod annotation manually) and it defines the sum of EPC resource requests of all containers in the pod.

We could have multiple sections in a VM but is there a need for it? The kernel EPC allocator is NUMA aware and checks the EPC section topology but the Kata VM is created without numa topology so only one section sounds justified.

Reply:
Yeah, makes sense -- just wanted to understand. Thanks Mikko!

Reply:
cool. One nit (sorry) - I wouldn't mention CLH here:

```suggestion
		// qemu sandboxes will only support one EPC per sandbox
		// this is because there is only one annotation (sgx.intel.com/epc)
		// to specify the size of the EPC.
```